src/compression.rs

```diff
@@ -0,0 +1,68 @@
+//! Compression strategy configs
+
+use lz4::block::CompressionMode;
+
+/// Wrapper of supported compression algorithms
+#[derive(Clone, Debug)]
+pub enum Compression {
```

Comment:
I am not sure if subtypes are allowing this, but to me the enum should be written like this:

```rust
# [derive(Clone, Debug, Default)]
pub enum Compression {
  # [default]
  None,
  Lz4 { mode: CompressionMode },
  Zlib { level: flate2::Compression },
  Zstd { level: i32 },
  Snappy,
}
```

Writing the enum like this will reduce a lot the code source as we do not need to create intermediate structures.

Reply:
Updated.